index.bs

```diff
@@ -1149,9 +1197,64 @@ The [=new=] {{MLGraphBuilder}} constructor steps are:
 1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, throw a "{{SecurityError}}" {{DOMException}} and abort these steps.
 1. Let |context| be the first argument.
 1. If the <a>validate MLContext</a> steps given |context| return `false`, throw a "{{TypeError}}" and abort these steps.
-
 1. Set {{MLGraphBuilder/[[context]]}} to |context|.
 
+### The `constant()` method  ### {#api-mlgraphbuilder-constant-method}
+Create a constant {{MLOperand}} that can be used in {{MLGraphBuilder}} methods.
+
+#### The {{MLGraphBuilder/constant(descriptor, bufferView)}} method  #### {#api-mlgraphbuilder-constant}
+<div class="note">
+    **Arguments:**
+        - *descriptor*: an {{MLOperandDescriptor}} object
+        - *bufferView*: an {{MLBufferView}}
+    **Returns:**: an {{MLOperand}} object.
+</div>
+<div algorithm=constant>
+    The {{MLGraphBuilder/constant(descriptor, bufferView)}} steps are:
+    <div class="note">
+        The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
+    </div>
+    1. Let |descriptor| be the first argument.
+    1. If |descriptor| is not an an object that [=implements=] {{MLOperandDescriptor}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+        1. If the [=byte length=] of |descriptor| is not supported by the underlying platform, then throw a "{{DataError}}" {{DOMException}} and stop.
```

Comment:
"byte length" algorithm may also need to be fixed for the case that `dimensions` is not present.

Reply:
OK, it seems that I missed if there was a convention that dimensions can be missing from descriptor.
Right now the Web IDL mandates the `descriptor` argument, and I inferred it needs both `type` and `dimensions`.
How should the algorithm behave if one or both are missing?

Reply:
@huningxin I think this is still open.

A spec should have clear instructions how implementations should handle the case `dimensions` (or `type`) is missing from a required `descriptor`.

Reply:
According to the current spec, `type` is required and `dimensions` is optional. If `dimensions` is not present, the `descriptor` represents a scalar value. It's fine to file an issue and fix it in the following PR.

Reply:
Would it make sense to assign a default value `0`, meaning scalar? Or is it better to handle with [=Map/exist=] and mention it means scalar?

Reply:
> is it better to handle with [=Map/exist=] and mention it means scalar?

This sounds good to me. Should we define a separate algorithm for this scalar check?

Reply:
OK, then it should be fine for now as we have separate algorithms. The `constant(descriptor, bufferView)` algorithm checks the `descriptor.dimensions`, but for the scalar case `constant(value, type)` it defines a new descriptor, currently saying the following:
```
 1. Let |descriptor| be a new {{MLOperandDescriptor}}.
        1. Set |descriptor|.{{MLOperandDescriptor/type}} to |type|.
        1. Set |descriptor|.{{MLOperandDescriptor/dimensions}} to `undefined`.
            <div class="note">
                In the case of a scalar constant, |descriptor|.{{MLOperandDescriptor/dimensions}} is ignored.
            </div>
```

That looks concise enough for me, but if you have suggestions to make it better, please tell.

Reply:
Yes, there are separate algorithms, we are fine now.